### PR TITLE
fix mp depckeck (#1887, #2092)

### PR DESCRIPTION
**Pull request by gfgtdf**
just like in the gui1 version opf mp create we just call a 'sync'
function whenever the user changed the selected era/scenario/mod. This
commit currently only fixes the check for era/scenario compability not
for mods.

it might also create an ininite recursion sometimes, have to investigate that.

### CODE REVIEW

**Comment by gfgtdf** [comment]
ok i just tried to test it, but the mp create screen still have other issues that need to be fixed before. 

1) when randomly selecting entried in the scnairo type menu button, the game often doesn't mange to show all of them

2) i randomly get 'manidity wml child missing ' error when i try to select games from that list.

**Comment by gfgtdf** [comment]
just fixed the second one in https://github.com/wesnoth/wesnoth/commit/6713070cc5a418515733131665e9315816d9e032 and didn't mange to repdoduce the first one.

**Comment by gfgtdf** [comment]
hmm it seems like https://github.com/wesnoth/wesnoth/issues/2194 shoudlo be fiuxed first before this one.

File: src/gui/dialogs/multiplayer/mp_create_game.cpp [hunk 1]
```diff
@@ -472,10 +513,15 @@ void mp_create_game::on_tab_select(window& window)
 	}
 }
 
-void mp_create_game::on_mod_toggle(const int index)
+void mp_create_game::on_mod_toggle(window& window, const int index, toggle_button* sender)
 {
-	create_engine_.set_current_mod_index(index);
-	create_engine_.toggle_current_mod();
+	(void)sender;
```
Comment (Vultraz) [inline src/gui/dialogs/multiplayer/mp_create_game.cpp]: Why is this here?

Comment (GregoryLundberg) [reply src/gui/dialogs/multiplayer/mp_create_game.cpp]: Looks like an unused parameter. So the question is why was the parameter needed. I'd assume to fix a required function signature.
